woocommerce/woocommerce
# [Experimental] Product Filters Overlay Navigation block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR adds a new block called `Navigation`, which is an inner block of `Product Filters Overlay`. The block contains a button to close the overlay. This PR only touches the block creation, editor settings, and editor preview, the frontend side of the block will be worked on in future PRs.

The Navigation block added in this PR:
- Can only be added inside `Product Filters Overlay` block.
- Can change the justification of inner text and icon.
- Can be sticky to the top of the overlay while scrolling.
- Styling:
  - By default, the block inherits the current theme style.
  - Typography, Color, Dimensions, and Border settings are available.
  - There are three navigation styles: Label and icon, Label only, and Icon only.
  - There are three button styles: Link, Fill, and Outline.
  - Icon size can be changed using a custom control.





Closes #47975 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

This should be tested by developers only.
0. Ensure the experimental blocks are enabled:
    - If you use the WooCommerce Beta Tester plugin, you can enable the experimental blocks in Tools > WCA Test Helper > Features. Toggle the `experimental-blocks` option.
    - If you build this PR locally, build the branch with `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build` command to have experimental blocks available.
1. Edit the Filters Overlay template part.
2. See the Navigation (Experimental) block.
3. Try changing settings and see the editor preview reflects those changes:
   - Alignment
   - Position
   - Color/Typo/Border/Dimensions.
   - Style, Button Style, Icon Size.
   - For Icon Size, verify the unit selector works as expected.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
